### ISSUE TITLE
Update 1.8.5 to 1.9.x: Projects cannot be added or deactivated

### ISSUE THREAD

**Issue by kuschel** [Baralga/baralga#62]
Hi, there is an error after updating from Baralga 1.8.5 to 1.9.x. Maybe it has something to do with change from number to UUID in project data. With imported data I'm not able to add more projects and to disable old ones.

Log snippet I found looks same in 1.9.0 and 1.9.1 like this:

```
2021-01-11 12:10:45,246 INFO  [main] org.remast.baralga.gui.BaralgaMain: Starting Baralga version 1.9.1.
2021-01-11 12:10:45,439 INFO  [main] org.remast.baralga.gui.BaralgaMain: Using application data directory "C:\Users\MyUserId\.Baralga".
2021-01-11 12:10:46,042 INFO  [main] org.remast.baralga.repository.file.BaralgaFileRepository: Using Baralga DB Version: 3, description: Migrated to uuids and added primary keys..
2021-01-11 12:11:08,187 ERROR [AWT-EventQueue-1] org.remast.swing.util.ExceptionUtils: Cannot add project without id.
java.lang.IllegalArgumentException: Cannot add project without id.
	at org.remast.baralga.repository.file.BaralgaFileRepository.addProject(BaralgaFileRepository.java:189)
	at org.remast.baralga.model.BaralgaDAO.addProject(BaralgaDAO.java:59)
	at org.remast.baralga.gui.model.PresentationModel.addProject(PresentationModel.java:189)
	at org.remast.baralga.gui.dialogs.ManageProjectsDialog.lambda$getAddProjectButton$0(ManageProjectsDialog.java:157)
	at javax.swing.AbstractButton.fireActionPerformed(Unknown Source)
	at javax.swing.AbstractButton$Handler.actionPerformed(Unknown Source)
	at javax.swing.DefaultButtonModel.fireActionPerformed(Unknown Source)
	at javax.swing.DefaultButtonModel.setPressed(Unknown Source)
	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(Unknown Source)
	at java.awt.AWTEventMulticaster.mouseReleased(Unknown Source)
	at java.awt.Component.processMouseEvent(Unknown Source)
	at javax.swing.JComponent.processMouseEvent(Unknown Source)
	at java.awt.Component.processEvent(Unknown Source)
	at java.awt.Container.processEvent(Unknown Source)
	at java.awt.Component.dispatchEventImpl(Unknown Source)
	at java.awt.Container.dispatchEventImpl(Unknown Source)
	at java.awt.Component.dispatchEvent(Unknown Source)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Unknown Source)
	at java.awt.LightweightDispatcher.processMouseEvent(Unknown Source)
	at java.awt.LightweightDispatcher.dispatchEvent(Unknown Source)
	at java.awt.Container.dispatchEventImpl(Unknown Source)
	at java.awt.Window.dispatchEventImpl(Unknown Source)
	at java.awt.Component.dispatchEvent(Unknown Source)
	at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.awt.EventQueue.access$500(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.awt.EventQueue$4.run(Unknown Source)
	at java.awt.EventQueue$4.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.awt.EventQueue.dispatchEvent(Unknown Source)
	at org.remast.swing.util.ExceptionUtils$ExceptionHandlingEventProcessor.dispatchEvent(ExceptionUtils.java:90)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.awt.WaitDispatchSupport$2.run(Unknown Source)
	at java.awt.WaitDispatchSupport$4.run(Unknown Source)
	at java.awt.WaitDispatchSupport$4.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.awt.WaitDispatchSupport.enter(Unknown Source)
	at java.awt.Dialog.show(Unknown Source)
	at java.awt.Component.show(Unknown Source)
	at java.awt.Component.setVisible(Unknown Source)
	at java.awt.Window.setVisible(Unknown Source)
	at java.awt.Dialog.setVisible(Unknown Source)
	at org.remast.baralga.gui.actions.ManageProjectsAction.actionPerformed(ManageProjectsAction.java:36)
	at javax.swing.AbstractButton.fireActionPerformed(Unknown Source)
	at javax.swing.AbstractButton$Handler.actionPerformed(Unknown Source)

```

**Comment by kuschel** [comment]
At the moment I think the migration resulted in data problems. Therefore further problems occurred. Creation of backup resulted in catastrophe: Most data in past was lost in backup (since 2013). That's why I recommend to start fresh after update to v 1.9.x. after backup of baralga.mv.db. I will do so :(

**Comment by da-blok** [comment]
The problem here ist that the project id is null, which was introduced in #60. I cannot imagine why this was done in that feature. 
Further changes in that commit:
There is also a comment removed which does not make any sense. 
For example: 
This line was remove:
`* @param id the unique id` but the parameter still exists in the constructor of Project.java

**Comment by remast** [comment]
I will look into it. The id of the project should be set in the class BaralgaFileRepository.

**Comment by remast** [comment]
Will be fixed in v1.9.2 that I'll release soon.
